std/assembly/util/string.ts

```diff
@@ -369,23 +369,46 @@ export function joinFloatArray<T>(dataStart: usize, length: i32, separator: stri
   return result;
 }
 
-export function joinStringArray(dataStart: usize, length: i32, separator: string): string {
+export function joinStringArray<T>(dataStart: usize, length: i32, separator: string): string {
   var lastIndex = length - 1;
   if (lastIndex < 0) return "";
-  if (!lastIndex) return load<string>(dataStart);
-
+  var value: T;
+  if (!lastIndex) {
+    if (isNullable<T>()) {
+      // @ts-ignore: type
+      return load<T>(dataStart)! || "";
```

Comment:
Looks like this aborts if the value is actually `null`?

Reply:
yeah, will remove `!`